spec.emu

```diff
@@ -1119,6 +1121,8 @@ contributors: Ron Buckton, Ecma International
               1. Set _completion_ to ThrowCompletion(_error_).
             1. Else,
               1. Set _completion_ to _result_.
+        1. NOTE: After _disposeCapability_ has been disposed, it will never be used again. The contents of _disposeCapability_.[[DisposableResourceStack]] can be discarded at this point.
+        1. Set _disposeCapability_.[[DisposableResourceStack]] to *undefined*.
```

Comment:
I'd set it to an empty list to avoid having to deal with a sentinel value everywhere else.

Reply:
> I'd set it to an empty list to avoid having to deal with a sentinel value everywhere else.

The upside of the sentinel value is that it makes things abundantly clear that a DisposeCapability is never intended to be reused, since we can assert on the absence of a list.

Reply:
You can also assert that the list is empty, no? In fact you can assert arbitrary English sentences, this isn't a program.

Reply:
Asserting the list is empty would break `DisposableStack`/`AsyncDisposableStack`. Asserting a clearly defined state avoids ambiguity for spec readers and implementers. i.e.,

> Assert: This DisposeCapability has never been passed as an argument to DisposeResources.

This seems way too ambiguous on its own.

Reply:
> Asserting the list is empty would break DisposableStack/AsyncDisposableStack

I don't follow, say more?

Reply:
`DisposableStack` and `AsyncDisposableStack` also use DisposeCapability, which starts with an empty list. This should not cause an Assert, or even a TypeError:

```js
const stack = new DisposableStack();
stack.dispose(); // DisposeCapability.[[DisposableResourceStack]] will be empty here.
```

In addition, `using x = null` does not add an entry to `DisposeCapability.[[DisposableResourceStack]]` either (only `await using x = null` does to ensure we trigger an Await), so an empty list is a perfectly valid state for a non-disposed DisposeCapability.

Reply:
But you don't even need these asserts if you use an empty list. AFAICT the reasoning chain goes: you want to editorially convey that the list doesn't keep things alive -> you null it out with a sentinel value -> you add asserts that the sentinel values don't flow into places it shouldn't.

I still don't see what's lost by changing the above to: you want to editorially convey that the list doesn't keep things alive -> you empty the list. You get the same editorial effect, and fewer chances of editorial errors by flowing the sentinel into wrong places.

Reply:
I'm concerned about the risk that a DisposeCapability is inadvertently reused. I am fine with changing this to just reset to an empty list, however.

Reply:
If you do want to use a sentinel value, you. might want to use something more evocative, like `~disposed~`.